### PR TITLE
Improve CI Setup

### PR DESCRIPTION
**Pull request by limdauto**
## Description

* Use the newer `cimg` image as opposed to the legacy `circleci`.
* Run `npm install` instead of `npm ci` to allow `node_modules` cache to take effect
* Remove conda in linux build. Just standard python venv is enough.
* Cache pip & python virtualenvs between builds.
* Parallelise `make pylint`

## Net result

* Linux build: 11m25s -> 7m35s (35%)
* Windows build: 11m47s -> 9m37s (18%)

Before:

Screenshot 2022-01-05 at 16 32 19

After:

Screenshot 2022-01-05 at 16 31 47

## Further improvement

Maybe we can run windows test on main only.

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,7 +43,7 @@ commands:
     steps:
       - run:
           name: Install Node dependencies
-          command: node tools/ci.js
+          command: npm install
```
Comment (tynandebold) [inline .circleci/config.yml:46]: I think we can delete the `tools/ci.js` file now. Doesn't seem to be used anywhere else.

**Review by tynandebold — APPROVED** [review]
Looks good to me. The results speak for themselves. Nice work.

Just one comment about the `tools/ci.js` file, but I'll approve straight away.

**Review by studioswong — COMMENTED** [review]
Nice one @limdauto - thanks for setting this up!

File: .circleci/config.yml [hunk 2]
```diff
@@ -6,36 +6,28 @@ orbs:
...
-            sudo rm -rf .pyenv/ /opt/circleci/.pyenv/
-            # Download and install miniconda
-            curl https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh  > miniconda.sh
-            bash miniconda.sh -b -p $HOME/miniconda
-            # Create an anaconda virtualenv for python ${CONDA_ENV_PY_VERSION} and make that the default python interpreter
-            echo ". /home/circleci/miniconda/etc/profile.d/conda.sh" >> $BASH_ENV
-            echo "conda deactivate; conda activate kedro-viz" >> $BASH_ENV
-            . /home/circleci/miniconda/etc/profile.d/conda.sh
-            conda create --name kedro-viz python=${CONDA_ENV_PY_VERSION} -y
-            source $BASH_ENV
```
Comment (studioswong) [inline .circleci/config.yml:34]: @limdauto this is more for my curiosity - I wonder why won't we need conda as the default python interpreter with the new setup? Is it becuase of the switch to use `cimg`?

**Pull request merged**